thewalrus/decompositions.py

```diff
@@ -75,26 +76,24 @@ def williamson(V, rtol=1e-05, atol=1e-08):
 
     M12 = np.real_if_close(sqrtm(V))
     Mm12 = np.linalg.inv(M12)
-    r1 = Mm12 @ omega @ Mm12
-    s1, K = schur(r1)
-    # In what follows a permutation matrix perm1 is constructed so that the Schur matrix has
+    Gamma = Mm12 @ omega @ Mm12
+    a, O = schur(Gamma)
+    # In what follows a permutation matrix perm is constructed so that the Schur matrix has
     # only positive elements above the diagonal
-    # Also the Schur matrix uses the x_1,p_1, ..., x_n,p_n  ordering thus a permutation perm2 is used
+    # Also the Schur matrix uses the x_1,p_1, ..., x_n,p_n  ordering thus the permutation perm is updated
```

Comment:
I wonder if here you could use `thewalrus.symplectic. xpxp_to_xxpp`

Reply:
yes, but then you would still need two permutations. In the current implementation we construct the permutation string and the permute once the array saving one permutation operation on the matrix.